### PR TITLE
HBASE-23675 Move to Apache parent POM version 22

### PR DESCRIPTION
**Pull request by petersomogyi**
Backport to branch-1.

### CODE REVIEW

**Comment by virajjasani** [comment]
Seems many pom changes so far are absent in branch-1 and with this, we will get them :)
+1 unless some unexpected error shows up in QA run. Hope assembly:single goal to build tarball is working fine with this change.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 42s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m 20s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m 56s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  compile  |   2m  5s |  branch-1 passed with JDK v1.7.0_242  |
| +1 :green_heart: |  shadedjars  |   3m 13s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   6m 22s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |  10m 54s |  branch-1 passed with JDK v1.7.0_242  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 34s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 54s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javac  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 17s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  javac  |   2m 17s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  xml  |   0m  6s |  The patch has 19 ill-formed XML file(s).  |
| +1 :green_heart: |  shadedjars  |   3m 33s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 26s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| -1 :x: |  javadoc  |   0m 17s |  hbase-common in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 13s |  hbase-procedure-jdk1.8.0_232 with JDK v1.8.0_232 generated 52 new + 0 unchanged - 0 fixed = 52 total (was 0)  |
| -1 :x: |  javadoc  |   0m 20s |  hbase-client in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 12s |  hbase-metrics-api in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 12s |  hbase-hadoop-compat-jdk1.8.0_232 with JDK v1.8.0_232 generated 100 new + 0 unchanged - 0 fixed = 100 total (was 0)  |
| -1 :x: |  javadoc  |   0m 11s |  hbase-metrics-jdk1.8.0_232 with JDK v1.8.0_232 generated 5 new + 0 unchanged - 0 fixed = 5 total (was 0)  |
| -1 :x: |  javadoc  |   0m 13s |  hbase-hadoop2-compat in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 12s |  hbase-prefix-tree in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 32s |  hbase-server in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 25s |  hbase-thrift in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 19s |  hbase-rest-jdk1.8.0_232 with JDK v1.8.0_232 generated 100 new + 0 unchanged - 0 fixed = 100 total (was 0)  |
| -1 :x: |  javadoc  |   0m 13s |  hbase-rsgroup in the patch failed with JDK v1.8.0_232.  |
| -1 :x: |  javadoc  |   0m 16s |  hbase-examples-jdk1.8.0_232 with JDK v1.8.0_232 generated 76 new + 0 unchanged - 0 fixed = 76 total (was 0)  |
| -1 :x: |  javadoc  |   0m 12s |  hbase-hbtop-jdk1.8.0_232 with JDK v1.8.0_232 generated 2 new + 0 unchanged - 0 fixed = 2 total (was 0)  |
| -1 :x: |  javadoc  |   0m 24s |  root in the patch failed with JDK v1.8.0_232.  |
| +1 :green_heart: |  javadoc  |  11m 28s |  the patch passed with JDK v1.7.0_242  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 159m 16s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   6m 30s |  The patch does not generate ASF License warnings.  |
|  |   | 239m 51s |   |


| Reason | Tests |
|-------:|:------|
| XML | Parsing Error(s): |
|   | hbase-assembly/pom.xml |
|   | hbase-client/pom.xml |
|   | hbase-common/pom.xml |
|   | hbase-examples/pom.xml |
|   | hbase-external-blockcache/pom.xml |
|   | hbase-hadoop-compat/pom.xml |
|   | hbase-hadoop2-compat/pom.xml |
|   | hbase-hbtop/pom.xml |
|   | hbase-it/pom.xml |
|   | hbase-metrics-api/pom.xml |
|   | hbase-metrics/pom.xml |
|   | hbase-prefix-tree/pom.xml |
|   | hbase-procedure/pom.xml |
|   | hbase-protocol/pom.xml |
|   | hbase-rest/pom.xml |
|   | hbase-rsgroup/pom.xml |
|   | hbase-server/pom.xml |
|   | hbase-thrift/pom.xml |
|   | pom.xml |
| Failed junit tests | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.mapreduce.TestLoadIncrementalHFiles |
|   | hadoop.hbase.mapreduce.TestLoadIncrementalHFilesUseSecurityEndPoint |
|   | hadoop.hbase.mapreduce.TestSecureLoadIncrementalHFiles |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.regionserver.TestCompactionState |
|   | hadoop.hbase.client.TestAdmin2 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1023 |
| Optional Tests | dupname asflicense javac javadoc unit shadedjars hadoopcheck xml compile |
| uname | Linux 6d8ecdff8490 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1023/out/precommit/personality/provided.sh |
| git revision | branch-1 / 248d80b |
| Default Java | 1.7.0_242 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_232 /usr/lib/jvm/zulu-7-amd64:1.7.0_242 |
| xml | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/xml.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-common-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/diff-javadoc-javadoc-hbase-procedure-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-client-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-metrics-api-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/diff-javadoc-javadoc-hbase-hadoop-compat-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/diff-javadoc-javadoc-hbase-metrics-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-hadoop2-compat-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-prefix-tree-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-server-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-thrift-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/diff-javadoc-javadoc-hbase-rest-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-hbase-rsgroup-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/diff-javadoc-javadoc-hbase-examples-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/diff-javadoc-javadoc-hbase-hbtop-jdk1.8.0_232.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-javadoc-root-jdk1.8.0_232.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/testReport/ |
| Max. process+thread count | 3903 (vs. ulimit of 10000) |
| modules | C: hbase-protocol hbase-common hbase-procedure hbase-client hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-prefix-tree hbase-server hbase-thrift hbase-rest hbase-rsgroup hbase-it hbase-examples hbase-external-blockcache hbase-hbtop hbase-assembly . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/1/console |
| versions | git=1.9.1 maven=3.0.5 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by petersomogyi** [comment]
> Hope assembly:single goal to build tarball is working fine with this change.

The tarball looks fine. I ran `mvn clean install -DskipTests site assembly:single -Prelease` and started hbase from tarball in standalone mode.

For the javadoc issue I'm uploading a second commit soon.

**Comment by petersomogyi** [comment]
From maven-javadoc-plugin 3.0.0 the build fails on JDK8. The configuration moved to a property.
With this the build shows this error message on JDK7 but but the build succeeds as before and there is not build failure on JDK8.
`[INFO] --- maven-javadoc-plugin:3.0.1:javadoc (default-cli) @ hbase-common ---
[WARNING] -Xdoclint:none option is not supported on Java version < 1.8. Ignore this option.`

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  12m 28s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   1m 49s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   7m 23s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m 52s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  compile  |   1m 59s |  branch-1 passed with JDK v1.7.0_242  |
| +1 :green_heart: |  shadedjars  |   3m 12s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   6m 22s |  branch-1 passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |  11m 14s |  branch-1 passed with JDK v1.7.0_242  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 36s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 14s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javac  |   2m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 33s |  the patch passed with JDK v1.7.0_242  |
| +1 :green_heart: |  javac  |   2m 33s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  xml  |   0m  6s |  The patch has 19 ill-formed XML file(s).  |
| +1 :green_heart: |  shadedjars  |   3m 28s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 31s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   6m 51s |  the patch passed with JDK v1.8.0_232  |
| +1 :green_heart: |  javadoc  |  10m 54s |  the patch passed with JDK v1.7.0_242  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 167m  9s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   6m 25s |  The patch does not generate ASF License warnings.  |
|  |   | 261m  9s |   |


| Reason | Tests |
|-------:|:------|
| XML | Parsing Error(s): |
|   | hbase-assembly/pom.xml |
|   | hbase-client/pom.xml |
|   | hbase-common/pom.xml |
|   | hbase-examples/pom.xml |
|   | hbase-external-blockcache/pom.xml |
|   | hbase-hadoop-compat/pom.xml |
|   | hbase-hadoop2-compat/pom.xml |
|   | hbase-hbtop/pom.xml |
|   | hbase-it/pom.xml |
|   | hbase-metrics-api/pom.xml |
|   | hbase-metrics/pom.xml |
|   | hbase-prefix-tree/pom.xml |
|   | hbase-procedure/pom.xml |
|   | hbase-protocol/pom.xml |
|   | hbase-rest/pom.xml |
|   | hbase-rsgroup/pom.xml |
|   | hbase-server/pom.xml |
|   | hbase-thrift/pom.xml |
|   | pom.xml |
| Failed junit tests | hadoop.hbase.mapreduce.TestSecureLoadIncrementalHFiles |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1023 |
| Optional Tests | dupname asflicense javac javadoc unit shadedjars hadoopcheck xml compile |
| uname | Linux 170a3f9cf314 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1023/out/precommit/personality/provided.sh |
| git revision | branch-1 / 248d80b |
| Default Java | 1.7.0_242 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_232 /usr/lib/jvm/zulu-7-amd64:1.7.0_242 |
| xml | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/2/artifact/out/xml.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/2/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/2/testReport/ |
| Max. process+thread count | 3986 (vs. ulimit of 10000) |
| modules | C: hbase-protocol hbase-common hbase-procedure hbase-client hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-prefix-tree hbase-server hbase-thrift hbase-rest hbase-rsgroup hbase-it hbase-examples hbase-external-blockcache hbase-hbtop hbase-assembly . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1023/2/console |
| versions | git=1.9.1 maven=3.0.5 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: pom.xml [hunk 1]
```diff
@@ -1374,6 +1344,9 @@
     <!-- Location of test resources -->
     <test.build.classes>${project.build.directory}/test-classes</test.build.classes>
     <shell-executable>bash</shell-executable>
+    <!-- TODO HBASE-15041 clean up our javadocs so jdk8 linter can be used.
+         property as of javadoc-plugin 3.0.0 -->
+    <doclint>none</doclint>
```
Comment (virajjasani) [inline pom.xml]: Ok and also we have this present in master and branch-2.
+1 from my side

**Comment by virajjasani** [comment]
cc: @apurtell @busbey

**Comment by petersomogyi** [comment]
Thanks for the reviews, I'll check the nightly.

**Pull request merged**